src/dc_receive_imf.rs

```diff
@@ -1304,11 +1297,7 @@ async fn calc_sort_timestamp(
         }
     }
 
-    if sort_timestamp >= dc_smeared_time(context).await {
-        sort_timestamp = dc_create_smeared_timestamp(context).await;
-    }
-
-    Ok(sort_timestamp)
+    Ok(sort_timestamp.min(dc_smeared_time(context).await))
```

Comment:
Yes, I think "or saved in the database" should be removed, there is nothing wrong with saving two equal timestamps for *received* messages. We only want to avoid sending messages with equal timestamps.